Added quotas for Kafka User (#2002)
strimzi/strimzi-kafka-operator: documentation/api/io.strimzi.api.kafka.model.KafkaUserQuotas.adoc

> Kafka cluster allows user to enforce certain requests to control usage of the resources by clients.

Reviewer: I wonder if this is a typo ... it doesn not seem to make sense to me.

```suggestion
Kafka cluster allows user to enforce certain quotas to control usage of the resources by clients.
```

> Kafka cluster allows user to enforce certain requests to control usage of the resources by clients.

Reviewer: Should we mention that we support only user based quotas and not client ID based quotas?

We can divide quotas to the two groups:

* network usage quotas, which as are defined as the byte rate threshold for each group of clients sharing a quota

> * hardware utilization quotas, which are defined as the percentage of time a client can utilize on request handler I/O threads and network threads of each broker within a quota window

Reviewer: Is it really _hardware utilization_? I would maybe just call it _utilization_ or maybe _CPU utilization_. Or is ths term from the KAfka docs?

There are several reasons why using quotas for the Kafka clients might be reasonable.
Let's consider wrongly set Kafka producer which is sending requests at high rate.
This can be considered as DoS attack and the client might get blocked.
With network limiting quota it is less probable to happen.

.An example Kafka user quotas
[source,yaml,subs=attributes+]
----
spec:
  quotas:

>     producerByteRate: 1000

Reviewer: Could we make the examples more realsitic? 1000 bytes per second seems not to be much.

Author: What value do you suggest? Official doc is not using high values neither (http://kafka.apache.org/documentation/#quotas). But it seems more reasonable to use _2^n_ values.

Reviewer: I don't know ... 1MB/s? 1048576?

> Kafka cluster allows user to enforce certain quotas to control usage of the resources by clients.

Reviewer 2: ```suggestion
Kafka allows the user to enforce certain quotas to control usage of resources by clients.
```

> We can divide quotas to the two groups:

Reviewer 2: ```suggestion
We can divide quotas to two groups:
```

* network usage quotas, which as are defined as the byte rate threshold for each group of clients sharing a quota
* CPU utilization quotas, which are defined as the percentage of time a client can utilize on request handler I/O threads and network threads of each broker within a quota window

There are several reasons why using quotas for the Kafka clients might be reasonable.

> Let's consider wrongly set Kafka producer which is sending requests at high rate.

Reviewer 2: ```suggestion
Consider a wrongly configured Kafka producer which is sending requests at too high a rate.
```

> This can be considered as DoS attack and the client might get blocked.

Reviewer 2: ```suggestion
This is misconfiguration can cause a denial of service to other clients and the misconfigured client ought to be blocked.
```

> With network limiting quota it is less probable to happen.

Reviewer 2: ```suggestion
Using a network limiting quota it is possible to prevent this from significantly impacting other clients.
```

> Currently we support only quotas configured for user level.

Reviewer 2: ```suggestion
Strimzi supports user-level quotas, but not client-level quotas.
```
